lib/apollo-federation/schema.rb

```diff
@@ -62,14 +74,44 @@ def original_query
         @orig_query_object || find_inherited_value(:original_query)
       end
 
+      def compose_directives
+        @compose_directives || find_inherited_value(:compose_directives, [])
+      end
+
+      def links
+        @links || find_inherited_value(:links, [])
+      end
+
+      def all_links
+        imported_directives = IMPORTED_DIRECTIVES
+        imported_directives += IMPORTED_DIRECTIVES_V2_1 if federation_2_1?
+        default_link = {
+          url: 'https://specs.apollo.dev/federation/v2.3',
+          import: imported_directives,
+        }
+        default_link[:as] = default_link_namespace if default_link_namespace != DEFAULT_LINK_NAMESPACE
+        [default_link, *links]
+      end
+
       def federation_2_prefix
-        federation_namespace = ", as: \"#{link_namespace}\"" if link_namespace != DEFAULT_LINK_NAMESPACE
+        schema = ['extend schema']
+
+        all_links.each do |link|
```

Comment:
This printing logic is fairly complex. Can we make it easier to read?

Reply:
Tried to do make it easier to read but not a ruby expert, please tell me if there's a better way :)